<?xml version="1.0" ?>
<sdf version="1.6">
    <model name="robobo">
        <static>0</static>

        <!-- LINKS -->
        <link name="body">
            <pose>-0.055984 0.034729 0.0308 0 0 0</pose>
            <inertial>
                <mass>0.48</mass>
                <inertia>
                    <ixx>0.000485229</ixx>
                    <ixy>0.000000058</ixy>
                    <ixz>0.000013696</ixz>
                    <iyy>0.001227181</iyy>
                    <iyz>-0.000000029</iyz>
                    <izz>0.001452813</izz>
                </inertia>
            </inertial>
            <collision name="body_collision">
                <geometry>
                	<mesh>
                		<uri>model://robobo/meshes/body_collision.dae</uri>
                	</mesh>
                </geometry>
                <surface>
                    <friction>
                        <ode>
                            <mu>0.2</mu>
                            <mu2>0.2</mu2>
                        </ode>
                    </friction>
                    <contact>
                        <poissons_ratio>0.46</poissons_ratio>
                        <elastic_modulus>7e8</elastic_modulus>
                        <ode>
                            <min_depth>0.0004</min_depth>
                        </ode>
                    </contact>
                </surface>
            </collision>
            <visual name="body_visual">
                <geometry>
                	<mesh>
                		<uri>model://robobo/meshes/body_visual.dae</uri>
                	</mesh>
                </geometry>
            </visual>
            <!--IR SENSORS-->
            <!-- vamos a borrar todo lo nuevo
            <sensor type="ray" name="front_c">
                <pose>-0.0803 0 0.0012 0 0 3.1415</pose>
                <visualize>false</visualize>
                <ray>
                    <scan>
                        <horizontal>
                            <samples>4</samples>
                            <min_angle>-0.2</min_angle>
                            <max_angle>0.17</max_angle>
                        </horizontal>
                        <vertical>
                            <samples>4</samples>
                            <min_angle>-0.12</min_angle>
                            <max_angle>0.16</max_angle>
                        </vertical>
                    </scan>
                <range>
                    <min>0</min>
                    <max>0.4</max>
                </range>
                </ray>
                <plugin name="infrared_range" filename="libinfrared_range.so">
                    <frameName> </frameName>
                    <fov>0</fov>
                </plugin>
            </sensor>
            <sensor type="ray" name="front_l">
                <pose>-0.0727 -0.026 0.0012 0 0.21 3.526</pose>
                <visualize>false</visualize>
                <ray>
                    <scan>
                        <horizontal>
                            <samples>4</samples>
                            <min_angle>-0.2</min_angle>
                            <max_angle>0.17</max_angle>
                        </horizontal>
                        <vertical>
                            <samples>4</samples>
                            <min_angle>-0.085</min_angle>
                            <max_angle>0.16</max_angle>
                        </vertical>
                    </scan>
                <range>
                    <min>0</min>
                    <max>0.4</max>
                </range>
                </ray>
                <plugin name="infrared_range" filename="libinfrared_range.so">
                    <frameName> </frameName>
                    <fov>0</fov>
                </plugin>
            </sensor>
            <sensor type="ray" name="front_ll">
                <pose>-0.0625 -0.0405 0.0012 0 0 3.927</pose>
                <visualize>false</visualize>
                <ray>
                    <scan>
                        <horizontal>
                            <samples>4</samples>
                            <min_angle>-0.2</min_angle>
                            <max_angle>0.17</max_angle>
                        </horizontal>
                        <vertical>
                            <samples>4</samples>
                            <min_angle>-0.12</min_angle>
                            <max_angle>0.16</max_angle>
                        </vertical>
                    </scan>
                <range>
                    <min>0</min>
                    <max>0.4</max>
                </range>
                </ray>
                <plugin name="infrared_range" filename="libinfrared_range.so">
                    <frameName> </frameName>
                    <fov>0</fov>
                </plugin>
            </sensor>
            <sensor type="ray" name="front_r">
                <pose>-0.0727 0.026 0.0012 0 0.21 2.758</pose>
                <visualize>false</visualize>
                <ray>
                    <scan>
                        <horizontal>
                            <samples>4</samples>
                            <min_angle>-0.2</min_angle>
                            <max_angle>0.17</max_angle>
                        </horizontal>
                        <vertical>
                            <samples>4</samples>
                            <min_angle>-0.085</min_angle>
                            <max_angle>0.16</max_angle>
                        </vertical>
                    </scan>
                <range>
                    <min>0</min>
                    <max>0.4</max>
                </range>
                </ray>
                <plugin name="infrared_range" filename="libinfrared_range.so">
                    <frameName> </frameName>
                    <fov>0</fov>
                </plugin>
            </sensor>
            <sensor type="ray" name="front_rr">
                <pose>-0.0625 0.0405 0.0012 0 0 2.356</pose>
                <visualize>false</visualize>
                <ray>
                    <scan>
                        <horizontal>
                            <samples>4</samples>
                            <min_angle>-0.2</min_angle>
                            <max_angle>0.17</max_angle>
                        </horizontal>
                        <vertical>
                            <samples>4</samples>
                            <min_angle>-0.12</min_angle>
                            <max_angle>0.16</max_angle>
                        </vertical>
                    </scan>
                <range>
                    <min>0</min>
                    <max>0.4</max>
                </range>
                </ray>
                <plugin name="infrared_range" filename="libinfrared_range.so">
                    <frameName> </frameName>
                    <fov>0</fov>
                </plugin>
            </sensor>
            <sensor type="ray" name="back_c">
                <pose>0.1201 0 0 0 0.21 0</pose>
                <visualize>false</visualize>
                <ray>
                    <scan>
                        <horizontal>
                            <samples>4</samples>
                            <min_angle>-0.2</min_angle>
                            <max_angle>0.17</max_angle>
                        </horizontal>
                        <vertical>
                            <samples>4</samples>
                            <min_angle>-0.085</min_angle>
                            <max_angle>0.16</max_angle>
                        </vertical>
                    </scan>
                <range>
                    <min>0</min>
                    <max>0.4</max>
                </range>
                </ray>
                <plugin name="infrared_range" filename="libinfrared_range.so">
                    <frameName> </frameName>
                    <fov>0</fov>
                </plugin>
            </sensor>
            <sensor type="ray" name="back_l">
                <pose>0.1055 -0.0308 0.001 0 0 -0.524</pose>
                <visualize>false</visualize>
                <ray>
                    <scan>
                        <horizontal>
                            <samples>4</samples>
                            <min_angle>-0.2</min_angle>
                            <max_angle>0.17</max_angle>
                        </horizontal>
                        <vertical>
                            <samples>4</samples>
                            <min_angle>-0.12</min_angle>
                            <max_angle>0.16</max_angle>
                        </vertical>
                    </scan>
                <range>
                    <min>0</min>
                    <max>0.4</max>
                </range>
                </ray>
                <plugin name="infrared_range" filename="libinfrared_range.so">
                    <frameName> </frameName>
                    <fov>0</fov>
                </plugin>
            </sensor>
            <sensor type="ray" name="back_r">
                <pose>0.1055 0.0308 0.001 0 0 0.524</pose>
                <visualize>false</visualize>
                <ray>
                    <scan>
                        <horizontal>
                            <samples>4</samples>
                            <min_angle>-0.2</min_angle>
                            <max_angle>0.17</max_angle>
                        </horizontal>
                        <vertical>
                            <samples>4</samples>
                            <min_angle>-0.12</min_angle>
                            <max_angle>0.16</max_angle>
                        </vertical>
                    </scan>
                <range>
                    <min>0</min>
                    <max>0.4</max>
                </range>
                </ray>
                <plugin name="infrared_range" filename="libinfrared_range.so">
                    <frameName> </frameName>
                    <fov>0</fov>
                </plugin>
            </sensor> 
        -->


        <!-- HASTA aqui IR -->
        </link> <!-- end of body link -->
        <link name="left_wheel">
            <pose>-0.082628 -0.038096 0.03225 0 0 0</pose>
            <inertial>
                <mass>0.046</mass>
                <inertia>
                    <ixx>0.000014552</ixx>
                    <ixy>0</ixy>
                    <ixz>0</ixz>
                    <iyy>0.000023921</iyy>
                    <iyz>0</iyz>
                    <izz>0.000014552</izz>
                </inertia>
            </inertial>
            <collision name="lw_collision">
                <pose>0 0 0 1.5708 0 0</pose>
                <geometry>
                    <cylinder>
                        <radius>0.03225</radius>
                        <length>0.026</length>
                    </cylinder>
                </geometry>
                <surface>
                    <friction>
                        <ode>
                            <mu>0.8</mu>
                            <mu2>0.8</mu2>
                        </ode>
                    </friction>
                    <contact>
                        <poissons_ratio>0.5</poissons_ratio>
                        <elastic_modulus>7e6</elastic_modulus>
                        <ode>
                            <min_depth>0.0008</min_depth>
                        </ode>
                    </contact>
                </surface>
            </collision>
            <visual name="lw_visual">
                <pose>0 -0.0024 0 0 0 0</pose>
                <geometry>
                	<mesh>
                		<uri>model://robobo/meshes/left_wheel_visual.dae</uri>
                	</mesh>
                </geometry>
            </visual>
        </link>
        <link name="right_wheel">
            <pose>-0.082628 0.10746 0.03225 0 0 0</pose>
            <inertial>
                <mass>0.046</mass>
                <inertia>
                    <ixx>0.000014552</ixx>
                    <ixy>0</ixy>
                    <ixz>0</ixz>
                    <iyy>0.000023921</iyy>
                    <iyz>0</iyz>
                    <izz>0.000014552</izz>
                </inertia>
            </inertial>
            <collision name="rw_collision">
                <pose>0 0 0 1.5708 0 0</pose>
                <geometry>
                    <cylinder>
                        <radius>0.03225</radius>
                        <length>0.026</length>
                    </cylinder>
                </geometry>
                <surface>
                    <friction>
                        <ode>
                            <mu>0.8</mu>
                            <mu2>0.8</mu2>
                        </ode>
                    </friction>
                    <contact>
                        <poissons_ratio>0.5</poissons_ratio>
                        <elastic_modulus>7e6</elastic_modulus>
                        <ode>
                            <min_depth>0.0008</min_depth>
                        </ode>
                    </contact>
                </surface>
            </collision>
            <visual name="rw_visual">
                <pose>0 0.0024 0 0 0 0</pose>
                <geometry>
                    <mesh>
                        <uri>
                            model://robobo/meshes/right_wheel_visual.dae
                        </uri>
                    </mesh>
                </geometry>
            </visual>
        </link>
        <link name="pan">
            <pose>-0.059743 0.034729 0.060151 0 0 3.14159</pose>
            <inertial>
                <mass>0.077</mass>
                <inertia>
                    <ixx>0.000037326</ixx>
                    <ixy>0</ixy>
                    <ixz>0</ixz>
                    <iyy>0.000037326</iyy>
                    <iyz>0</iyz>
                    <izz>0.000074536</izz>
                </inertia>
            </inertial>
            <collision name="pan_collision">
                <geometry>
                    <cylinder>
                        <radius>0.044</radius>
                        <length>0.003</length>
                    </cylinder>
                </geometry>
            </collision>
            <visual name="pan_visual">
                <geometry>
                    <mesh>
                        <uri>
                            model://robobo/meshes/pan_visual.dae
                        </uri>
                    </mesh>
                </geometry>
            </visual>
        </link>
        <link name="tilt_smartphone">
            <pose>-0.149486 0.03499 0.0843 0 0 3.14159</pose>
            <inertial>
                <mass>0.18</mass>
                <inertia>
                    <ixx>0.000091682</ixx>
                    <ixy>0</ixy>
                    <ixz>0</ixz>
                    <iyy>0.000375824</iyy>
                    <iyz>0</iyz>
                    <izz>0.000465727</izz>
                </inertia>
            </inertial>
            <self_collide>1</self_collide>
            <collision name="tilt_collision">
                <geometry>
                    <box>
                        <size>0.1581 0.0778 0.0077</size>
                    </box>
                </geometry>
            </collision>
            <visual name="tilt_visual">
                <pose>-0.079749 -0.002249 -0.004338 0 0 0</pose>
                <geometry>
                    <mesh>
                        <uri>
                            model://robobo/meshes/tilt_smartphone_visual.dae
                        </uri>
                    </mesh>
                </geometry>
            </visual>


            <!-- Camara 
            <sensor type="camera" name="front_camera">
                <pose>0.0695 0.0105 0.004 0 -1.5708 3.1415</pose>
                <visualize>false</visualize>
                <camera>
                    <image>
                        <width>480</width>
                        <height>640</height>
                    </image>
                </camera>
                <plugin name="camera_controller1" filename="librobobo_camera_utils.so"/>
                <plugin name="camera_controller2" filename="librobobo_camera.so">
                    <frameName>Camera</frameName>
                </plugin>
            </sensor> -->
            <!-- IMU -
            <sensor name="imu_sensor" type="imu">
                <visualize>false</visualize>
                <plugin name="imu_plugin" filename="librobobo_imu_sensor.so">
                    <topicName>imu</topicName>
                    <bodyName>imu_link</bodyName>
                    <updateRateHZ>10.0</updateRateHZ>
                    <gaussianNoise>0.0</gaussianNoise>
                    <xyzOffset>0 0 0</xyzOffset>
                    <rpyOffset>0 0 0</rpyOffset>
                    <frameName>imu_link</frameName>
                </plugin>
            </sensor> -->
        </link>

        <!-- JOINTS -->
        <joint name="left_motor" type="revolute">
            <parent>body</parent>
            <child>left_wheel</child>
            <pose>0 0 0 0 0 0</pose>
            <axis>
                <xyz>0 -1 0</xyz>
                <dynamics>
                    <friction>0.022</friction>
                </dynamics>
            </axis>
        </joint>
        <joint name="right_motor" type="revolute">
            <parent>body</parent>
            <child>right_wheel</child>
            <pose>0 0 0 0 0 0</pose>
            <axis>
                <xyz>0 -1 0</xyz>
                <dynamics>
                    <friction>0.022</friction>
                </dynamics>
            </axis>
        </joint>
        <joint name="pan_motor" type="revolute">
            <parent>body</parent>
            <child>pan</child>
            <pose>0 0 0 0 0 0</pose>
            <axis>
                <xyz>0 0 -1</xyz>
                <dynamics>
                    <friction>0.15</friction>
                </dynamics>
            </axis>
        </joint>
        <joint name="tilt_motor" type="revolute">
            <parent>pan</parent>
            <child>tilt_smartphone</child>
            <pose>-0.089743 0 -0.005 0 0 0</pose>
            <axis>
                <xyz>0 -1 0</xyz>
                <dynamics>
                    <friction>0.22</friction>
                </dynamics>
            </axis>
        </joint>

        <!-- PLUGINS OF MODEL-->
        <plugin name="encoders" filename="libencoders.so"/>
        <plugin name="move_wheels" filename="libmove_wheels.so"/>
        <plugin name="move_pan_tilt" filename="libmove_pan_tilt.so"/>

    </model>
</sdf>
